io/src/lib.rs

```diff
@@ -230,6 +230,59 @@ impl<T: AsRef<[u8]>> BufRead for Cursor<T> {
     }
 }
 
+/// A [`Read`]er which keeps an internal buffer to avoid hitting the underlying stream directly for
+/// every read, implementing [`BufRead`].
+pub struct BufReader<'a, R: Read> {
+    inner: &'a mut R,
```

Comment:
Oh, wait, my questions about the reference weren't addressed. https://github.com/rust-bitcoin/rust-bitcoin/pull/3162#pullrequestreview-2235205845